base: Fix Integer overflow in AddrRange

Harshil2107 opened the pull request:
This PR fixes the bug mentioned in #240.

Change-Id: I32139fa37b741185ba766d49dfa8cdab34f19010

Harshil2107 committed:
base: fix Integer overflow in AddrRange bug

An issue raised in #240 where if an address range ends
at the last byte of a 64 bit address space, it will be
considered a subset of any other address range that starts
at the first byte of the range.

Change-Id: I517f4717052eda2504de971be0eb59ee9a623dd3

pranith commented on src/base/addr_range.test.cc:
```diff
@@ -1544,6 +1544,19 @@ TEST(AddrRangeTest, SubtractionAssignmentOfRangeListFromRangeList)
 
+TEST(AddrRangeTest, isNotSubsetLastByte)
+{
+    /* An issue raised in https://github.com/gem5/gem5/issues/240 where if an
```
This comment can be copied into the commit message to make the motivation clearer.

Harshil2107 committed:
base: Add a check for edge case

- Now check for the condition where the bigger address range wraps but smaller does not.

Change-Id: Icc7a549afaf82a277dc2845255aa1702a1d662e0
